pkg/tfbridge/schema.go

```diff
@@ -480,8 +484,20 @@ func MakeTerraformOutput(v interface{},
 			return resource.NewComputedProperty(
 				resource.Computed{Element: resource.NewStringProperty("")})
 		}
-		// Else it's just a string.
-		return resource.NewStringProperty(t)
+
+		// Is there a schema available to us? If not, it's definitely just a string.
+		if tfs == nil {
+			return resource.NewStringProperty(t)
+		}
+
+		// Otherwise, it might be a string that needs to be coerced to match the Terraform schema type. Coerce the
+		// string to the Go value of the correct type and, if the coercion produced something different than the string
+		// value we already have, re-make the output.
+		coerced, err := CoerceTerraformString(tfs.Type, t)
+		if err != nil || coerced == t {
+			return resource.NewStringProperty(t)
+		}
+		return MakeTerraformOutput(coerced, tfs, ps, assets, rawNames)
```

Comment:
FWIW, I think this could have just been `resource.NewPropertyValue(coerced)`

Reply:
oh, nice - I did not know that existed.